clang/unittests/Basic/DarwinSDKInfoTest.cpp

```diff
@@ -78,6 +78,43 @@ TEST(DarwinSDKInfo, VersionMappingParseError) {
           .has_value());
 }
 
+TEST(DarwinSDKInfo, SystemPrefix) {
+  llvm::json::Object SDKSettings({{"CanonicalName", "macosx26.0"},
+                                  {"Version", "26.0"},
+                                  {"MaximumDeploymentTarget", "26.0.99"}});
+  llvm::json::Object SupportedTargets;
+  llvm::json::Object MacOS({{"Archs", {"x86_64", "arm64"}},
+                            {"LLVMTargetTripleVendor", "apple"},
+                            {"LLVMTargetTripleSys", "macos"},
+                            {"LLVMTargetTripleEnvironment", ""},
+                            {"SystemPrefix", "/System/macOSSupport"}});
+  llvm::json::Object MacCatalyst({{"Archs", {"x86_64", "arm64"}},
+                                  {"LLVMTargetTripleVendor", "apple"},
+                                  {"LLVMTargetTripleSys", "ios"},
+                                  {"LLVMTargetTripleEnvironment", "macabi"},
+                                  {"SystemPrefix", "/System/iOSSupport"}});
+  llvm::json::Object DriverKit({{"Archs", {"x86_64", "arm64"}},
+                                {"LLVMTargetTripleVendor", "apple"},
+                                {"LLVMTargetTripleSys", "driverkit"},
+                                {"LLVMTargetTripleEnvironment", ""},
+                                {"SystemPrefix", ""}});
+  SupportedTargets["macosx"] = std::move(MacOS);
+  SupportedTargets["iosmac"] = std::move(MacCatalyst);
+  SupportedTargets["driverkit"] = std::move(DriverKit);
+  SDKSettings["SupportedTargets"] = std::move(SupportedTargets);
+
+  auto SDKInfo = DarwinSDKInfo::parseDarwinSDKSettingsJSON(&SDKSettings);
+  ASSERT_TRUE(SDKInfo);
+  EXPECT_EQ(SDKInfo->getSystemPrefix(Triple("arm64-apple-macos26.0")),
+            "/System/macOSSupport");
+  EXPECT_EQ(SDKInfo->getSystemPrefix(Triple("ppc-apple-macos26.0")),
+            "/System/macOSSupport");
+  EXPECT_EQ(SDKInfo->getSystemPrefix(Triple("arm64-apple-ios26.0")),
+            "/System/macOSSupport");
```

Comment:
Does `ios` result in  `"/System/macOSSupport"` because there's no entry for it in `SDKInfo` and it defaults to use the `CanonicalName` as the platform value? Why is that desired behavior? This seems like it would result in an observable change for users that may not be holding the SDK arguments in the _common_ way for compiler invocations.

Reply:
Yes that's why. I did that because so far there aren't any SDKs that have multiple system prefixes, or any that have a system prefix and also content in the SDK root. So it seemed more convenient that you basically always get the canonical system prefix, and you can always `-nostdlibinc`or `-nostdinc` if you really don't want it. But if you think it would be better to get nothing for triples that aren't explicitly listed then that's an easy change.

Reply:
It seems to me the goal of this PR is that there should be no observable change for clients that have well behaving configured projects. From a stability standpoint, I think it's better to preserve as much of the preexisting behavior for existing workflows as possible to land this change. If any client needs to make changes to get the right system prefixes, that should be handled separately from this pr.